src/js/elements/menu.js

```diff
@@ -1,174 +1,194 @@
-        for (const idx in indicators) {
-            let indicator = indicators[idx];
-            count += subindicatorsInIndicator(indicator);
-        }
+      for (const idx in indicators) {
+        let groups = indicators[idx].metadata.groups;
+        count += this._subindicatorsInIndicator(groups);
+      }
     }
 
     return count;
-}
+  }
+
+  checkIfSubCategoryHasChildren(detail) {
+    let hasChildren = false;
+    for (const [_, data] of Object.entries(detail.indicators)) {
+      if (!hasChildren && typeof data.child_data !== 'undefined') {
+        for (const [_, arr] of Object.entries(data.child_data)) {
+          hasChildren = hasChildren || arr.length > 0;
+        }
+      }
+    }
+
+    return hasChildren;
+  }
+
+  checkIfCategoryHasChildren(detail) {
+    for (const [_, subcategoryDetail] of Object.entries(detail.subcategories)) {
+      if (this.checkIfSubCategoryHasChildren(subcategoryDetail))
+        return true
+    }
+
+    return false
+  }
 
-function subindicatorsInIndicator(indicator) {
-    return indicator.metadata.groups.length;
 }
 
+
 // TODO this entire file needs to be refactored to use thhe observer pattern
-export function loadMenu(dataMapperMenu, data, subindicatorCallback) {
-    function addSubIndicators(wrapper, category, subcategory, indicator, groups, indicators, choropleth_method, indicatorId) {
-
-        $(".data-category__h3", wrapper).remove();
-        $(".data-category__h4", wrapper).remove();
-
-        if (groups !== null && typeof groups.subindicators !== 'undefined') {
-            groups.subindicators.forEach((subindicator) => {
-                const newSubIndicatorElement = indicatorItemTemplate.cloneNode(true);
-                $(".truncate", newSubIndicatorElement).text(subindicator);
-                $(newSubIndicatorElement).attr('title', subindicator);
-
-                wrapper.append(newSubIndicatorElement);
-
-                const parents = {
-                    category: category,
-                    subcategory: subcategory,
-                    indicator: indicator
-                }
-
-                $(newSubIndicatorElement).on("click", (el) => {
-                    setActive(el);
-                    if (subindicatorCallback != undefined)
-                        subindicatorCallback({
-                            el: el,
-                            data: data,
-                            indicatorTitle: indicator,
-                            selectedSubindicator: subindicator,
-                            indicators: indicators,
-                            parents: parents,
-                            choropleth_method: choropleth_method,
-                            indicatorId: indicatorId
-                        })
-                });
-            });
+export class DataMapperMenu extends Component {
+  constructor(parent) {
+    super(parent);
+    this._data = null;
+    this._subindicatorCallback = null;
+  }
+
+  init(data, subindicatorCallback) {
```

Comment:
This seems awkward - see my comment later about removing this method

Reply:
I did not see any other comment regarding this. Did you perhaps forget to add it?